package-type-check/pkg/checkers/virtual.go

```diff
@@ -0,0 +1,79 @@
+package checkers
+
+import (
+	"fmt"
+	"os/exec"
+
+	"github.com/chainguard-dev/cg-tw/package-type-check/pkg/utils"
+)
+
+// Check performs the virtual package checks
+func CheckVirtualPackage(pkg string, virtualPkgs []string) error {
```

Comment:
I'm a bit confused by the implementation of the logic on this check, because it doesn't seem to follow the one we already had in the shell-based pipeline.

Reply:
No it dose not, I will put a comment explaining my issue with the script based pipeline we have, and why I think this is better.

Reply:
ok!